roles/kubernetes_setup/tasks/main.yml

```diff
@@ -2,119 +2,117 @@
 #
 #  This role sets up kubernetes apiserver, controller-mgr, and scheduler pods
 #
-  - name: Determine kubelet version
-    shell: kubelet --version | awk '{print $2}' | cut -d 'v' -f 2
-    register: kv
+- name: Get kubernetes version
+  include_role:
+    name: rpm_version
+  vars:
+    rv_rpms: kubernetes-node
 
-  - name: Set kubelet version fact
-    set_fact:
-      kubelet_version: "{{ kv.stdout }}"
+- name: Define kube 1.5 vars
+  set_fact:
+    kube_maj_ver: "1.5"
+    kube_apiserver: "registry.access.redhat.com/rhel7/kubernetes-apiserver"
+    kube_controller_manager: "registry.access.redhat.com/rhel7/kubernetes-controller-mgr"
+    kube_scheduler: "registry.access.redhat.com/rhel7/kubernetes-scheduler"
+  when: atomic_host['kubernetes-node'] | version_compare('1.6', '<')
 
-  - name: Define kube 1.5 vars
-    set_fact:
-      kube_maj_ver: "1.5"
-      kube_apiserver: "registry.access.redhat.com/rhel7/kubernetes-apiserver"
-      kube_controller_manager: "registry.access.redhat.com/rhel7/kubernetes-controller-mgr"
-      kube_scheduler: "registry.access.redhat.com/rhel7/kubernetes-scheduler"
-    when: kubelet_version | version_compare('1.6', '<')
+- name: Define kube 1.6 images
+  set_fact:
+    kube_maj_ver: "1.6"
+    kube_apiserver: "registry.fedoraproject.org/f26/kubernetes-apiserver"
+    kube_controller_manager: "registry.fedoraproject.org/f26/kubernetes-controller-manager"
+    kube_scheduler: "registry.fedoraproject.org/f26/kubernetes-scheduler"
+  when: atomic_host['kubernetes-node'] | version_compare('1.6', '>=')
 
-  - name: Define kube 1.6 images
-    set_fact:
-      kube_maj_ver: "1.6"
-      kube_apiserver: "registry.fedoraproject.org/f26/kubernetes-apiserver"
-      kube_controller_manager: "registry.fedoraproject.org/f26/kubernetes-controller-manager"
-      kube_scheduler: "registry.fedoraproject.org/f26/kubernetes-scheduler"
-    when: kubelet_version | version_compare('1.6', '>=')
+- name: pull kubernetes api-server, controller-mgr, scheduler
+  command: docker pull {{ item }}
+  register: pull
+  failed_when: item not in pull.stdout
+  with_items:
+    - "{{ kube_apiserver }}"
+    - "{{ kube_controller_manager }}"
+    - "{{ kube_scheduler }}"
 
-  - name: pull kubernetes api-server, controller-mgr, scheduler
-    command: docker pull {{ item }}
-    register: pull
-    failed_when: item not in pull.stdout
-    with_items:
-      - "{{ kube_apiserver }}"
-      - "{{ kube_controller_manager }}"
-      - "{{ kube_scheduler }}"
+- name: make db directory
+  file: path=/etc/kubernetes/manifests state=directory mode=0755
```

Comment:
Since this role is getting modified, do you want to change this task to use the format found elsewhere?

```
- name: make db directory
  file:
    path: /etc/kubernetes/manifests
    state: directory
    mode: 0755
```

Reply:
Deleted two spaces from the beginning of each line now I have to do all the cleanup work. haha